### PR TITLE
Add swagger json and UI to application

### PR DESCRIPTION
**Pull request by jamielennox**
We're going to use this application as the base of a take home front-end
test in which the applicant is expected to develop a new front end for
the api. This means we need to much better document the API.

Add a swagger generator and all the tagging required to make that work,
and a swagger UI so that we can point users at the interface for the API
documentation.

### CODE REVIEW

File: ui/index.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+func assetHandler(cfg Config) http.Handler {
+	// so not secure!
+	return http.FileServer(cfg.Assets)
+}
+
+func uiHandler(cfg Config, router *mux.Router) {
+	router.PathPrefix("/js/").Handler(assetHandler(cfg))
+	router.PathPrefix("/css/").Handler(assetHandler(cfg))
+	router.PathPrefix("/images/").Handler(assetHandler(cfg))
+	router.PathPrefix("/swagger/").Handler(assetHandler(cfg))
```
Comment (TWinsnes) [inline ui/index.go]: Why did you move these into a separate file, while leaving the health check where it is? Makes more sense to me to leave the global routes in`ui.go`, while index specific entries in `index.go`

Comment (jamielennox) [reply ui/index.go]: I was thinking of all these routes as ui/frontend routes as separate from the api routes. But the main file is already called ui, so I couldn't call it that. Healthcheck is neither frontend nor backend so stayed in the main.

I don't mind though, i'll move them back.

**Comment by TWinsnes** [comment]
Could you bump the version on this as well? So we trigger a new release

File: ui/index.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
+// Copyright © 2018 Thomas Winsnes <tom@vibrato.com.au>
```
Comment (chrisfowles) [inline ui/index.go]: this license should probably be switched over to Vibrato... assuming @TWinsnes is ok with that.

Comment (TWinsnes) [reply ui/index.go]: I'm okay with that
